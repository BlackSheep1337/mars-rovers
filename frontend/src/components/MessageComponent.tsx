import { MessageProps } from "../types";

const MessageComponent: React.FC<MessageProps> = ({ message }) => {
  if (!message.text) return null;

  return (
    <p className={`text-lg mt-4 ${message.type === "error" ? "text-red-600" : "text-green-600"}`}>
      {message.text}
    </p>
  );
};

export default MessageComponent;