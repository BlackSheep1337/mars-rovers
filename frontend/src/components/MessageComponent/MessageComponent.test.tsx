import { render, screen } from "@testing-library/react";
import MessageComponent from "./MessageComponent";

describe("MessageComponent", () => {
  test("does not render when message.text is empty", () => {
    render(<MessageComponent message={{ text: "", type: "success" }} />);
    expect(screen.queryByText(/.+/)).not.toBeInTheDocument();
  });

  test("renders message text when message.text is provided", () => {
    render(<MessageComponent message={{ text: "Operation successful", type: "success" }} />);
    expect(screen.getByText("Operation successful")).toBeInTheDocument();
  });

  test("applies text-red-600 class for error messages", () => {
    render(<MessageComponent message={{ text: "An error occurred", type: "error" }} />);
    const message = screen.getByText("An error occurred");
    expect(message).toHaveClass("text-red-600");
  });

  test("applies text-green-600 class for success messages", () => {
    render(<MessageComponent message={{ text: "Operation successful", type: "success" }} />);
    const message = screen.getByText("Operation successful");
    expect(message).toHaveClass("text-green-600");
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<MessageComponent message={{ text: "Snapshot test", type: "success" }} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
