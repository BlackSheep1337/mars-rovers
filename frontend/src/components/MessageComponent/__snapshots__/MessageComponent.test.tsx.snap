// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageComponent matches snapshot 1`] = `
<DocumentFragment>
  <p
    class="text-lg mt-4 text-green-600"
  >
    Snapshot test
  </p>
</DocumentFragment>
`;
